jetty-core/jetty-http2/jetty-http2-server/src/main/java/org/eclipse/jetty/http2/server/AuthorityCustomizer.java

```diff
@@ -0,0 +1,63 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http2.server;
+
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.util.HostPort;
+import org.eclipse.jetty.util.URIUtil;
+
+/**
+ * <p>A {@link HttpConfiguration.Customizer} that synthesizes the authority when the
+ * {@link HttpHeader#C_AUTHORITY} header is missing.</p>
+ * <p>After customization, the synthesized authority is accessible via
+ * {@link HttpURI#getAuthority()} from the {@link Request} object.</p>
+ * <p>The authority is synthesized from the {@code Host} header.
+ * If the {@code Host} header is also missing, it is synthesized using
+ * {@link Request#getServerName(Request)} and {@link Request#getServerPort(Request)}.</p>
+ */
+public class AuthorityCustomizer implements HttpConfiguration.Customizer
```

Comment:
If someone has both this, and `HostHeaderCustomizer` active at the same time, what happens?

Reply:
Let me try answer this one to see if I understand correctly.  Two options:

 a) `HostHeaderCustomizer` then `AuthorityCustomizer` - The authority, if missing will be set from the configured host values if missing from the request.
 b) `AuthorityCustomizer` then `HostHeaderCustomizer` - Any missing host header will be constructed from the authority created by the Authority customizer (probably from the connection meta data)

Reply:
@gregw is correct, the 2 customizers are orthogonal.